### DISCUSSION TITLE
I need to reset my DB because my migrations are unhappy.  Will I lose my authentication though?

### DISCUSSION THREAD

**Discussion by AdventureBear** [prisma/prisma#23042] [Q&A]
### Question

So this totally could have been user error, but I made an update to my schema, and the migration that was created tried to edit the new model before the table was created. So now I cant figure out how to recover the sync and make it happy. I did all the SQL updates I needed manually (supabase).   

I also took the time this evenign to write a seed file that is working, so i can get rid of all my existing data...but will this also wipe out my authentication? It's google auth and was painful to set up (it was my first time). 

This migration not being in sync is a real drag, but it seems to be common thta people have to rest the db. 

BEfore I do it I just want to know what will happen to my authentication stuff, and if I'll have to keep redoing that, what's the easiest way ? 

Thanks for any help. 

### How to reproduce (optional)

run reset db command

### Expected behavior (optional)

user authentication tables are preserved

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @AdventureBear 👋 

Thank you for raising this question.

Resetting the database will wipe out all the data, including your authentication data. If you have a seed file you've confirmed is working, you can use it to repopulate your database after the reset. However, if your seed file does not include your authentication setup, you may need to redo that manually.

Before you go ahead with resetting the database, you should address the root cause of the failed migration. If the issue was due to a change in the Prisma schema that cannot be applied to a database with data, you can:

- Delete the migration.sql file.
- Modify the schema - for example, add a default value to the mandatory field.
- Run the migration again using prisma migrate dev.

Can you try the steps to see if it helps fix your issue?

**Comment by AdventureBear** [comment]
Yes, I tried all of those steps.

I posted an issue about this, and as far as I can tell, the root cause was
because the migration file tried to update a table that hadn’t been created
yet. I placed a new table in the schema, and I change the name of one of
the prior ENUMs.

this could’ve been user error, and I will probably do fewer changes at one
time in the future, prima should’ve created the new table 1st, and then
change the name of the field contained within it.

I tried doing what you describe, and the migration still cannot be applied.

I tried combination of things, including the migration entry in the
database. But in the end, I just manually reset it and created a seed file.

Interestingly, it still shows me logged in and gives me access to all the
pages, even though my authentication is gone.

Thanks for the reply.

**Reply by ludralph** [reply]
Thanks for confirming the issue is resolved.
